plugins/new-relic/retention.yaml

```diff
@@ -265,31 +264,27 @@ presetScripts:
       def add_source_dest_columns(df):
         df.pod = df.ctx['pod']
         df.namespace = df.ctx['namespace']
+        df.container = df.ctx['container']
```

Comment:
I think we need to move these three lines down, inserting them right above `df = add_source_dest_columns(df)`

The plugin doesn't report a compiler error for this (and appears to ship records successfully for the oazizi dev cluster), but if you copy this script into the scratchpad and remove the px.plugin.start_time and add a px.display() call, then you get the following error: `Compiler error on line 23, column 20: UDF 'upid_to_pod_name' must execute before blocking nodes such as limit, agg, and join..`

Reply:
This is the script I'm using in the Scratch Pad: 
```
#px:set max_output_rows_per_table=500

import px

def remove_duplicate_traces(df):
    ''' Removes duplicate traces.
        For historical reasons, Pixie traces MySQL requests on both the client AND server side:
        https://github.com/pixie-io/pixie/blob/5e5598ac46f39219148a36468b5318b1466a92d4/src/stirling/source_connectors/socket_tracer/conn_tracker.cc#L639
    '''
    # Keep client-side traces if server is outside the cluster (can't be resolved to pod or svc)
    df.remote_pod_id = px.ip_to_pod_id(df.remote_addr)
    df.remote_service_id = px.ip_to_service_id(df.remote_addr)
    df.remote_outside_cluster = df.remote_pod_id == '' and df.remote_service_id == ''
    df_client_traces = df[df.trace_role == 1 and df.remote_outside_cluster]
    df_server_traces = df[df.trace_role == 2]
    df_server_traces.append(df_client_traces)
    return df_server_traces

def remove_ns_prefix(column):
    return px.replace('[a-z0-9\-]*/', column, '')

def add_source_dest_columns(df):
    df.pod = df.ctx['pod']
    df.namespace = df.ctx['namespace']
    df.container = df.ctx['container']
    # If remote_addr is a pod, get its name. If not, use IP address.
    df.ra_pod = px.pod_id_to_pod_name(px.ip_to_pod_id(df.remote_addr))
    df.is_ra_pod = df.ra_pod != ''
    df.ra_name = px.select(df.is_ra_pod, df.ra_pod, df.remote_addr)
    df.is_server_tracing = df.trace_role == 2
    df.is_source_pod_type = px.select(df.is_server_tracing, df.is_ra_pod, True)
    df.is_dest_pod_type = px.select(df.is_server_tracing, True, df.is_ra_pod)
    # Set client and server based on trace_role.
    df.source_pod = px.select(df.is_server_tracing, df.ra_name, df.pod)
    df.destination_pod = px.select(df.is_server_tracing, df.pod, df.ra_name)
    df.source_service = px.pod_name_to_service_name(df.source_pod)
    df.destination_service = px.pod_name_to_service_name(df.destination_pod)
    df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
    df.source_namespace = px.pod_name_to_namespace(df.source_pod)
    df.source_service = remove_ns_prefix(df.source_service)
    df.destination_service = remove_ns_prefix(df.destination_service)
    df.source_container = px.select(df.is_server_tracing, '', df.container)
    df.source_pod = remove_ns_prefix(df.source_pod)
    df.destination_pod = remove_ns_prefix(df.destination_pod)
    # If the destination service is missing then try the nslookup
    df.destination_service = px.select(
        df.destination_service != '',
        df.destination_service,
        px.nslookup(df.remote_addr),
    )
    # If the destination service is still missing then set the remote_addr
    df.destination_service = px.select(
        df.destination_service != '',
        df.destination_service,
        df.remote_addr,
    )
    return df.drop(['ra_pod', 'is_ra_pod', 'ra_name', 'is_server_tracing'])

df = px.DataFrame('mysql_events', start_time='-5m')
df = remove_duplicate_traces(df)
df = add_source_dest_columns(df)
df.normed_query_struct = px.normalize_mysql(df.req_body, df.req_cmd)
df.query = px.pluck(df.normed_query_struct, 'query')
df = df[df.query != ""]
df.start_time = df.time_ - df.latency
df.latency = df.latency / (1000 * 1000)
df.req_bytes = px.length(df.req_body)
df.resp_bytes = px.length(df.resp_body)
df.req_cmd = px.mysql_command_name(df.req_cmd)
df.cluster_name = px.vizier_name()
df.cluster_id = px.vizier_id()
df.pixie = 'pixie'
df.db_system = 'mysql'

px.display(df)
```

Reply:
I'm going to file a separate bug for this weird behavior.